### DISCUSSION TITLE
How to encode messages?

### DISCUSSION THREAD

**Discussion by lz1998** [tokio-rs/prost#525] [Q&A] [answered]
```rust
            let mut buf: Vec<u8> = Vec::with_capacity(200);
            frame.encode(&mut buf).unwrap();
```
Frame is generated from .proto file.

![image](https://user-images.githubusercontent.com/9082086/131242183-051b8c07-8687-4d14-922d-c50889f9e8e3.png)

**Comment by fd** [comment, marked as answer]
Did you bring the `prost::Message` trait in scope? Try adding a `use prost::Message;`.

**Comment by alanhe** [comment]
Hi @fd,

Could you give me some performance advices?

My CPU profile shows that, it consumes a lot CPU when I call `Message::encode_to_vec` to serialize a protobuf object.

From the flame graph, `Message::encoded_len` and `core::ptr::drop_inplace` take long time.

What I want is to make HTTP requests via `reqwest`. I serialize the protobuf object to be the HTTP body.

Am I doing it right? Are there more efficient ways of do this?

![Snip20220304_27](https://user-images.githubusercontent.com/5449446/156744066-6ec856af-57df-4f9a-b8fb-46ee939d319a.png)

**Reply by alanhe** [reply]
BTW, I'm using prost 0.9.0.

**Reply by alanhe** [reply]
Sorry, could you help? @LucioFranco

**Reply by LucioFranco** [reply]
This isn't the right place to ask this question, you can open a separate discussion to look into it.

**Reply by alanhe** [reply]
https://github.com/tokio-rs/prost/discussions/602  Thanks.
